### PR TITLE
Worker fix for it crashing on wrong binary path in Manager config

### PR DESCRIPTION
**Pull request by binarypearl**
The Worker will crash if the path in the Manager sqlite database is incorrect.  Basically the Worker will run fine until it gets a job, then it crashes with an 'OSError: [Errno 2] No such file or directory' or  'OSError: [Errno 13] Permission denied'.  

I fixed this by modifying the run_blender_in_thread(options) routine in worker/application/controllers.py.  I added a try/except clause starting at the 'PROCEES = subprocess.Popen(stuff, stuff, stuff)', and ending at the end of the routine.

If we get an OSError, it will say that it couldn't find the path to the Blender binary, and it will show the path that it tried to reference.  

The Worker will continue to run and won't crash now.  Once the Manager configuration is updated with the correct path, the Worker will pick it up and continue rendering.

---

I tend to be verbose with my comments when making changes in code.  I also tag them with my initials and date for record keeping.  But if you would like this done differently, I'm more than happy to change it around. 

--Shaun

### CODE REVIEW

**Comment by binarypearl** [comment]
I checked out the version from master and re-did just what is needed (thus wiping out my comments/debug statements).

As far as where the except goes, I tried doing the except earlier, but the overall Worker still crashed.

Basically when I put the except righter after PROCESS, it errored out with: "AttributeError: 'NoneType' object has no attribute 'stdout'"

Slightly after that, the PROCESS code wouldn't error out, and it would display the print() statements in the except clause.  But the Worker would still crash later with things like:  "UnboundLocalError: local variable 'full_output' referenced before assignment" and ""UnboundLocalError: local variable 'time_init' referenced before assignment" as I moved the except further down the code.

Basically variable scope at that point.  So it wasn't until the very end of the routine where the overall Worker stopped crashing.  So I left the 'except OSError' code in the same spot as my original commit.

--Shaun

**Comment by peppelinux** [comment]
Could be usefull to catch the exception and write, in stdout, that your main blender interpreter were not found? This could prevent some brain crushing to final user.

**Comment by binarypearl** [comment]
Sorry that I have been inactive lately.  I'm in the middle of switching jobs and relocating to Washington DC.  

If I fix the conflicts that are now in the file, are you still fine conceptually with my attempt to fix this issue?  (peppelinux's last comment is what I am trying to do).

--Shaun

**Comment by peppelinux** [comment]
Talking with Francesco it seems to me that the sooner prioriorities will be:
- user management (oriented on Flask-Security)

I think that the main part of our milestone could, or will, be described on:
https://flamenco.readme.io/
https://flamenco.readme.io/blog/

I understand your doubts, any blindfolded pilot should drive a plane. And this is what a programmer is, a pilot. After blenderconf I'm sure that coding will start again.

I also noticied that there will be another flamenco.io server implementation written in Cpp, but never used.

**Comment by binarypearl** [comment]
Sounds good, please keep me in mind when development starts up again!

--Shaun

**Comment by fsiddi** [comment]
I'm currently still very busy with other development work, but I'll be back on Flamenco soon.

**Comment by sybrenstuvel** [comment]
Thanks for your pull request. I'm closing this on, though, since it relates to the old, now-obsolete version of Flamenco. Read about the new Flamenco 2.0 at https://flamenco.io/.
